sssd_test_framework/roles/generic.py

```diff
@@ -1362,3 +1409,187 @@ def lockout(self, duration: int, attempts: int) -> GenericPasswordPolicy:
         :rtype: GenericPasswordPolicy
         """
         pass
+
+
+class GenericDNSServer(ABC, BaseObject):
+    """
+    DNS management utilities.
+    """
+
+    @abstractmethod
+    def zone(self, name: str) -> GenericDNSZone:
+        """
+        Get GenericDNSZone object.
+
+        :param name: Zone name.
+        :type name: str
+        :return: GenericDNSZone object.
+        :rtype: GenericDNSZone
+        """
+        pass
+
+    @abstractmethod
+    def get_forwarders(self) -> list[str]:
+        """
+        Get DNS global forwarders.
+
+        :return:  List of forwarders.
+        :rtype: list[str]
+        """
+        pass
+
+    @abstractmethod
+    def add_forwarders(self, forwarders: list[str]) -> GenericDNSServer:
+        """
+        Set DNS forwarders.
+
+        :param forwarders: List of DNS servers, optional
+        :type forwarders: list[str] | None = None
+        :return: Self.
+        :rtype: GenericDNSServer
+        """
+        pass
+
+    @abstractmethod
+    def remove_forwarders(self, ip_address: str) -> None:
+        """
+        Remove DNS server forwarders.
+
+        :param ip_address: IP address.
+        :type ip_address: str
+        """
+        pass
+
+    @abstractmethod
+    def list_zones(self) -> list[str]:
+        """
+        List all DNS zones.
+        """
+        pass
+
+    @abstractmethod
+    def record_exists(self, fqname: str, ipv6: bool | None = False) -> bool:
```

Comment:
Same as above comment, why not just use `get_record` to check if a record exists?